index.js

```diff
@@ -1,34 +1,53 @@
-import makeTemplate from "lodash.template";
-import { visitParents as visit } from "unist-util-visit-parents";
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @package rehype-lodash-template
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ * @copyright Viktor Yakubiv & Contributors
+ * @license MIT
+ */
 
-const attach = (options = {}) => {
-  const { values = {}, templateSettings } = options;
+import makeTemplate from "lodash.template";
+import { visitParents } from "unist-util-visit-parents";
 
-  const cache = new Map();
+/**
+ * A rehype plugin to replace template strings based on lodash.template.
+ *
+ * @since 0.0.0
+ * @author Viktor Yakubiv (www.yakubiv.com)
+ *
+ * @param {object} param0 { values, templateSettings }
+ * @returns Rehype Plugin
+ */
+export default function attach({ values, templateSettings }) {
+  const valueCache = new Map();
 
-  const template = (str) => {
-    if (!cache.has(str)) {
-      cache.set(str, makeTemplate(str, templateSettings));
+  const getTemplate = (identifier) => {
+    if (!valueCache.has(identifier)) {
+      valueCache.set(identifier, makeTemplate(identifier, templateSettings));
     }
 
-    return cache.get(str);
+    return valueCache.get(identifier);
   };
 
-  const substitute = (str) => {
-    const newString = template(str)(values);
+  const substitute = (identifier) => {
+    const template = getTemplate(identifier);
+    const substitutedString = template(values);
 
     // When the template matches one value completely, it's better to return
     // original value so the compiler can process it in own way.
     // This is helpful for boolean attributes like `hidden`
-    // and useful for nulish values to remove the attribute or text completely.
+    // and useful for nullish values to remove the attribute or text completely.
     const valueEntry = Object.entries(values).find(
-      ([name, value]) => str.indexOf(name) >= 0 && newString === String(value)
+      ([name, value]) =>
+        identifier.indexOf(name) >= 0 && substitutedString === String(value)
     );
 
-    return valueEntry?.[1] ?? newString;
+    return valueEntry?.[1] ?? substitutedString;
   };
 
-  const visitor = (node, ancestors) => {
+  const nodeVisitor = (node, ancestors) => {
```

Comment:
There is only one visitor, and renaming it is just writing more words for words. Operate on context: rehype is a _syntax tree_; tree consists of _nodes_; nodes can be visited by a _visitor_ function. It's clear that this function is a `[tree] [node] visitor`.

Unfortunately, some languages suffer from this extra verbosity by design but this is not about JavaScript.